content/post/lessons-learned-building-an-accessible-web-application-framework/index.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "Lessons learned building an accessible web application framework"
+authors: ["Matt Henkes"]
+date: 2020-09-21
+tags: [accessibility, terra, engineering, oss]
+---
+
+Since 2017, We at Cerner have been building Terra, a single page web application framework, and accessibility has always been a primary goal. Over the years we've learned a lot about how to build and test accessible websites, as well as the diverse group of people we're serving. With this blog post, we'd like to take a moment to share what we've learned so far.
+
+## Why accessibility matters
+
+At Cerner, we build software for many different users in many different roles. Doctors and nurses may first come to mind as the primary users for our EHR, but hospitals aren't made up of providers alone and neither are our users. We also provide applications for administrative users and with the Healthe portal app, our users are the general patient population. Naturally, with such a broad user base a percentage will require assistive technologies to be able to use our applications. To be as inclusive as possible, Cerner applications must be as accessible as possible, and the Terra Application Framework is one way we are working towards that goal.
+
+## What does it mean for a site to be accessible
+
+I'll be honest, when we started in 2017, I thought creating an accessible website simply meant appropriately applying ARIA labels to DOM elements. In the last few years, we've learned, with the help of our accessibility researchers, that creating an accessible site is so much more than that. So, let's hop in our time machine and discuss how we should have approached accessibility with appropriate hindsight (foresight? Time travel is confusing).
+
+### Identify how users interact with our site with accessible technologies
+
+As we discussed above, our broad user base includes a diverse array of people and an equally diverse array of accessibility needs. When discussing web accessibility, it's easy to assume that means ensuring a screen reader appropriately can navigate and read your site. Screen reader compatibility is important but if your user has impaired mobility, but fine eyesight, you're not really helping them. We've compiled the following use cases to ensure we are covering all expected interactions and needs.
+
+* Impaired vision
+* Impaired mobility
+* Impaired cognition
+
+Additionally, individuals could temporarily fall into any of the above categories through injury, device capability, location, or conditions.
+
+### Create areas of focus for each use case
+
+For each of the above use cases we've identified the appropriate assistive technologies to focus on.
```

Comment:
> For each of the above use cases we've identified the appropriate assistive technologies to focus on.

Maybe add a comma and avoid ending on a preposition:

> For each of the above use cases, we've identified the appropriate assistive technologies to utilize.

Reply:
updated